lib/core/device.js

```diff
@@ -254,38 +254,35 @@ device.detectUsb = function(driverId, cb) {
     return cb(null, devdata);
   };
 
-  if (self._os === 'win') {
-    var chromeVersion = /Chrome\/([0-9]+)/.exec(navigator.userAgent)[1];
-    self.log('Chrome version detected as', chromeVersion);
-
-    chrome.serial.getDevices(function(results) {
-      if (chromeVersion >= 50) {
-        // As of Chrome 50, product ID and vendor ID is returned
-        // on Windows (only) for serial.getDevices
-        // Chromium bug report: https://bugs.chromium.org/p/chromium/issues/detail?id=599101
-        results = _.filter(results, function(p) {
+
+  var chromeVersion = /Chrome\/([0-9]+)/.exec(navigator.userAgent)[1];
+  self.log('Chrome version detected as', chromeVersion);
+
+  chrome.serial.getDevices(function(results) {
+    if (chromeVersion >= 50) {
+      // As of Chrome 50, product ID and vendor ID is returned
+      // Chromium bug report: https://bugs.chromium.org/p/chromium/issues/detail?id=599101
+      results = _.filter(results, function(p) {
+        if (self._os === 'win') {
           return (p.vendorId === driverManifest.vendorId) &&
           (p.productId === driverManifest.productId);
-        });
-      }
-      else{
-        // With Chrome 49 and earlier, there may be
-        // multiple connected devices
-        self.log('Connected device(s): ');
-        for(var i in results) {
-            self.log(results[i].path);
+        } else {
+          return (p.vendorId === driverManifest.vendorId) &&
+          (p.productId === driverManifest.productId) &&
+          p.path.match('/dev/cu.+');
```

Comment:
On Mac and Linux, each serial device shows up twice in `/dev`, one for `tty` (incoming) and one for `cu` (outgoing) communication, so you can do full duplex. For example, both `/dev/cu.usbserial` and `/dev/tty.usbserial` will show up. We always want to connect to the `/dev/cu*` one.